### PR TITLE
lightning 2.1.0

### PR DESCRIPTION
**Pull request by bfontaine**

### CODE REVIEW

File: Library/Formula/lightning.rb [hunk 1]
```diff
@@ -13,8 +11,41 @@ class Lightning < Formula
     sha1 "eb8db8f32ceddaddaf445bb45268d13d753f89ba" => :lion
   end
 
+  option "with-check", "run the test suit when building"
```
Comment (mikemcquaid) [inline Library/Formula/lightning.rb]: `test suite`

Comment (bfontaine) [inline Library/Formula/lightning.rb]: Ah thanks, fixed!

File: Library/Formula/lightning.rb [hunk 2]
```diff
@@ -13,8 +11,41 @@ class Lightning < Formula
...
   end
 
+  option "with-check", "run the test suite when building"
+
   def install
-    system "./configure","--disable-dependency-tracking", "--prefix=#{prefix}"
-    system "make install"
+    system "./configure", "--disable-dependency-tracking",
+                          "--prefix=#{prefix}"
+    system "make", "check" if build.with? "check"
```
Comment (jacknagel) [inline Library/Formula/lightning.rb]: How long does this take? Should we just run it unconditionally?

Comment (bfontaine) [inline Library/Formula/lightning.rb]: On my machine it takes 18 seconds without the check and 40 seconds with it.

Comment (jacknagel) [inline Library/Formula/lightning.rb]: I guess I am pretty skeptical about adding "with-check" options. Do people actually use them? What good is it if we never run it in CI? Given we have bottles, I don't seem the harm in running it unconditionally.

Comment (mikemcquaid) [inline Library/Formula/lightning.rb]: > I guess I am pretty skeptical about adding "with-check" options. Do people actually use them? What good is it if we never run it in CI? Given we have bottles, I don't seem the harm in running it unconditionally.

:+1:

Comment (bfontaine) [inline Library/Formula/lightning.rb]: Ok, I removed the option.

**Comment by tdsmith** [comment]
Test suite fails on each platform: `FAIL: float.nodata`

**Comment by bfontaine** [comment]
Mmmmh added `--disable-disassembler` to `./configure` to remove the `binutils` dependency, it now works on Mavericks (but fails on ML and Yosemite).

**Comment by xu-cheng** [comment]
This is what I got when testing it locally.
```
$ cat ~/Library/Logs/Homebrew/lightning/config.log | grep error
clang: error: unsupported option '-V -isystem/usr/include/libxml2'
clang: error: unknown argument: '-qversion'
conftest.c:11:10: fatal error: 'ac_nonexistent.h' file not found
1 error generated.
conftest.c:11:10: fatal error: 'ac_nonexistent.h' file not found
1 error generated.
clang: error: linker command failed with exit code 1 (use -v to see invocation)
| /* Override any GCC internal prototype to avoid an error.
```

**Comment by bfontaine** [comment]
I get the same output. The two `ac_nonexistent.h` errors are [normal](http://www.linuxquestions.org/questions/linux-software-2/prelink-configuration-error-fatal-error-ac_nonexistent-h-no-such-file-or-direct-939408/#post4651030), and the option and argument clang errors are [also normal](http://www.postgresql.org/message-id/14172.1420767633@sss.pgh.pa.us). This leaves us with:

    clang: error: linker command failed with exit code 1 (use -v to see invocation)

Which expanded gives:

```
configure:11615: checking for mremap
configure:11615: clang -o conftest -g -O2   conftest.c  >&5
Undefined symbols for architecture x86_64:
  "_mremap", referenced from:
      _main in conftest-be4a17.o
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

I guess that’s a normal behavior since `./configure` tests a lot of things, some of which might not be supported on the platform.

**Comment by tdsmith** [comment]
Checked debian to see if they specced any interesting dependencies; they don't, but they mothballed the package a few years ago: https://packages.qa.debian.org/l/lightning/news/20110207T223156Z.html

**Comment by bfontaine** [comment]
I’m not familiar with MacPort but [they don’t seem to specify any dependency](https://trac.macports.org/browser/trunk/dports/devel/lightning/Portfile) either.

**Comment by bfontaine** [comment]
Fedora, which uses the 2.1.0 release, [requires](http://pkgs.fedoraproject.org/cgit/lightning.git/tree/lightning.spec) `texinfo` (provided by OS X), `binutils-devel` (the Homebrew formula is called `binutils`), and, if I understand correctly, `prelink` (don’t know what this is).

**Comment by jacknagel** [comment]
When I see successful compilation but failing tests, I try compiling with `ENV.O0` to rule out optimization bugs.

**Comment by bfontaine** [comment]
Thanks @jacknagel, trying this.

**Comment by jacknagel** [comment]
I'll look at this locally tonight.

**Comment by jacknagel** [comment]
Looks like a race condition in `make check`, `make check -j1` works for me locally.

**Comment by jacknagel** [comment]
For reference, I figured this out because `check/test-suite.log` contained:

```
===============================================
   GNU lightning 2.1.0: check/test-suite.log
===============================================

# TOTAL: 97
# PASS:  96
# SKIP:  0
# XFAIL: 0
# FAIL:  1
# XPASS: 0
# ERROR: 0

.. contents:: :depth: 2

FAIL: float.nodata
==================

diff: float.out: No such file or directory
rm: float.out: No such file or directory

```

**Comment by bfontaine** [comment]
That was it, thanks @jacknagel!

**Pull request closed without merging**